lib/archethic/contracts/interpreter/function_interpreter.ex

```diff
@@ -47,12 +47,31 @@ defmodule Archethic.Contracts.Interpreter.FunctionInterpreter do
   end
 
   @doc """
-  Execute function code and returns the result
+  Execute public function code and returns the result
+
+  Raise on SC's errors and timeout
   """
   @spec execute(ast :: any(), constants :: map(), args_names :: list(), args_ast :: list()) ::
           result :: any()
   def execute(ast, constants, args_names \\ [], args_ast \\ []) do
-    Scope.execute(ast, constants, args_names, args_ast)
+    task =
```

Comment:
**Suggestion**:
Since the Interpreter is meant to interpret, I think it should not take decision about rules to execute functions.
This timeout should be handled by the root level of smart contract in `Contracts` module